oremda/registry.py

```diff
@@ -0,0 +1,79 @@
+class Registry:
+    def __init__(self, client):
+        self.client = client
+        self.images = {}
+        self.run_kwargs = {}
+
+    def _inspect(self, image_name, prefix='oremda', delimiter='.'):
+        image = self.client.image(image_name)
+
+        labels = image.oremda_labels
+
+        ports = labels.setdefault('ports', {})
+        ports.setdefault('input', {})
+        ports.setdefault('output', {})
```

Comment:
We can move this code into the `oremda_labels` property if you think that would simplify things.